File: Shared/Palette.swift
import SwiftUI

enum Palette {
    /// Clawd's own color.
    static let clay = Color(red: 215 / 255, green: 119 / 255, blue: 87 / 255)
    static let clayLight = Color(red: 236 / 255, green: 150 / 255, blue: 118 / 255)
    static let clayDeep = Color(red: 190 / 255, green: 92 / 255, blue: 62 / 255)

    static let amber = Color(red: 245 / 255, green: 178 / 255, blue: 76 / 255)
    static let amberDeep = Color(red: 222 / 255, green: 140 / 255, blue: 44 / 255)
    static let red = Color(red: 255 / 255, green: 105 / 255, blue: 94 / 255)
    static let redDeep = Color(red: 226 / 255, green: 62 / 255, blue: 58 / 255)

    static let sweat = Color(red: 140 / 255, green: 204 / 255, blue: 255 / 255)
    static let heart = Color(red: 255 / 255, green: 110 / 255, blue: 128 / 255)

    static let backgroundTop = Color(red: 52 / 255, green: 43 / 255, blue: 39 / 255)
    static let backgroundBottom = Color(red: 29 / 255, green: 25 / 255, blue: 23 / 255)

    static var background: LinearGradient {
        LinearGradient(colors: [backgroundTop, backgroundBottom], startPoint: .top, endPoint: .bottom)
    }

    static func fill(for severity: UsageLimit.Severity) -> LinearGradient {
        let colors: [Color] = switch severity {
        case .normal: [clayDeep, clayLight]
        case .warning: [amberDeep, amber]
        case .critical: [redDeep, red]
        }
        return LinearGradient(colors: colors, startPoint: .leading, endPoint: .trailing)
    }

    static func tint(for severity: UsageLimit.Severity) -> Color {
        switch severity {
        case .normal: clayLight
        case .warning: amber
        case .critical: red
        }
    }
}

File: windows/Clawdmeter/Usage/Palette.cs
using System.Windows.Media;

namespace Clawdmeter.Usage;

public static class Palette
{
    /// Clawd's own color.
    public static readonly Color Clay = Color.FromRgb(215, 119, 87);
    public static readonly Color ClayLight = Color.FromRgb(236, 150, 118);
    public static readonly Color ClayDeep = Color.FromRgb(190, 92, 62);

    public static readonly Color Amber = Color.FromRgb(245, 178, 76);
    public static readonly Color AmberDeep = Color.FromRgb(222, 140, 44);
    public static readonly Color Red = Color.FromRgb(255, 105, 94);
    public static readonly Color RedDeep = Color.FromRgb(226, 62, 58);

    public static readonly Color Sweat = Color.FromRgb(140, 204, 255);
    public static readonly Color Heart = Color.FromRgb(255, 110, 128);
    public static readonly Color Spark = Color.FromRgb(255, 214, 140);

    public static readonly Color BackgroundTop = Color.FromRgb(52, 43, 39);
    public static readonly Color BackgroundBottom = Color.FromRgb(29, 25, 23);

    public static LinearGradientBrush Background()
    {
        var brush = new LinearGradientBrush(BackgroundTop, BackgroundBottom, 90);
        brush.Freeze();
        return brush;
    }

    public static LinearGradientBrush Fill(Severity severity)
    {
        var (from, to) = severity switch
        {
            Severity.Warning => (AmberDeep, Amber),
            Severity.Critical => (RedDeep, Red),
            _ => (ClayDeep, ClayLight),
        };
        var brush = new LinearGradientBrush(from, to, 0);
        brush.Freeze();
        return brush;
    }

    public static Color Tint(Severity severity) => severity switch
    {
        Severity.Warning => Amber,
        Severity.Critical => Red,
        _ => ClayLight,
    };

    public static SolidColorBrush Brush(Color color)
    {
        var brush = new SolidColorBrush(color);
        brush.Freeze();
        return brush;
    }
}
